core/search/bm25.hpp

```diff
@@ -69,7 +69,12 @@ class BM25 final : public irs::ScorerBase<BM25, BM25Stats> {
   static void init();  // for trigering registration in a static build
 
   BM25(float_t k = K(), float_t b = B(), bool boost_as_score = BOOST_AS_SCORE())
-  noexcept : k_{k}, b_{b}, boost_as_score_{boost_as_score} {}
+    : k_{k}, b_{b}, boost_as_score_{boost_as_score} {
+    if (k < 0.f || b < 0.f || 1.f < b) {
+      IRS_LOG_WARN(
```

Comment:
Please remove this warning or reduce it to TRACE. It will pollute logs after update of existing instances with scorers violating this requirement.

Reply:
It's intentionally, because user shouldn't use invalid score